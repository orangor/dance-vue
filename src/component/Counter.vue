<!--
 * @Descripttion: 
 * @version: 
 * @Author: luohongwen
 * @Date: 2020-07-30 14:11:38
 * @LastEditTime: 2020-07-30 14:11:49
--> 
<template>
    <div>
        {{ count }}
        <button @click="increment">自增</button>
    </div>
</template>

<script>
export default {
    data() {
        return {
            count: 0,
        }
    },

    methods: {
        increment() {
            this.count++
        },
    },
}
</script>