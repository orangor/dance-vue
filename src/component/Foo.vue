<!--
 * @Descripttion: 
 * @version: 
 * @Author: luohongwen
 * @Date: 2020-08-07 10:23:39
 * @LastEditTime: 2020-08-07 10:24:41
-->
<template>
    <div>
        <transition>
            <p v-if="show">Foo</p>
        </transition>
    </div>
</template>

<script>
export default {
    data() {
        return {
            show: true,
        }
    },
}
</script>