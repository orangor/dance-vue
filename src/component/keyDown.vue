<!--
 * @Descripttion: 
 * @version: 
 * @Author: luohongwen
 * @Date: 2020-08-07 13:43:07
 * @LastEditTime: 2020-08-07 14:12:48
-->
<template>
    <input type="text" @keydown.prevent="onKeydown" v-model="quantity" />
</template>

<script>
const KEY_DOWN = 40
const KEY_UP = 38
const ESCAPE = 27

export default {
    data() {
        return {
            quantity: 0,
        }
    },

    methods: {
        increment() {
            this.quantity += 1
        },
        decrement() {
            this.quantity -= 1
        },
        clear() {
            this.quantity = 0
        },
        onKeydown(e) {
            if (e.keyCode === ESCAPE) {
                this.clear()
            }
            if (e.keyCode === KEY_DOWN) {
                this.decrement()
            }
            if (e.keyCode === KEY_UP) {
                this.increment()
            }
            if (e.key === 'a') {
                this.quantity = 13
            }
        },
    },

    watch: {
        quantity: function (newValue) {
            this.$emit('input', newValue)
        },
    },
}
</script>