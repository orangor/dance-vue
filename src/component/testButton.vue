<!--
 * @Descripttion: 
 * @version: 
 * @Author: luohongwen
 * @Date: 2020-08-07 11:20:06
 * @LastEditTime: 2020-08-07 14:05:06
-->
<template>
    <div>
        <button class="yes" @click="callYes">Yes</button>
        <button class="no" @click="callNo">No</button>
    </div>
</template>

<script>
export default {
    name: 'testButton',

    props: {
        callMe: {
            type: Function,
        },
    },

    methods: {
        callYes() {
            this.callMe('yes')
        },
        callNo() {
            this.callMe('no')
        },
    },
}
</script>