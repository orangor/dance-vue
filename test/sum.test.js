/*
 * @Descripttion:
 * @version:
 * @Author: luohongwen
 * @Date: 2020-07-17 12:35:35
 * @LastEditTime: 2020-08-07 17:20:29
 */
import Vue from 'vue';
import ElementUI from 'element-ui';
import functions from '../src/sum.js';
import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import Counter from '../src/component/Counter.vue'
import ChildComponent from '../src/component/Dance.vue'
import ParentComponent from '../src/component/App.vue'
import Foo from '../src/component/Foo.vue'
import testButton from '../src/component/testButton.vue'
import keyDown from '../src/component/keyDown.vue'
import mocks from '../src/component/mock.vue'
import sinon from 'sinon'
import flushPromises from 'flush-promises'
import VueRouter from 'vue-router'
/*
describe('Counter.vue', () => {
    it('increments count when button is clicked', async () => {
        const wrapper = shallowMount(Counter)
        wrapper.find('button').trigger('click')
        await Vue.nextTick()
        expect(wrapper.find('div').text()).toMatch('1')
    })
})

describe('Conter', () => {
    it('renders a div-color', () => {
        const wrapper = mount(Counter, {
            propsData: {
                color: 'red'
            }
        })
        expect(wrapper.props().color).toBe('red')
    })
})
describe('Conter', () => {
    const wrapper = mount(Counter)
    it('renders the correct markup', () => {
        expect(wrapper.html()).toContain('<span class="count">0</span>')
    })
    //检查已存在的元素
    it('has a button', () => {
        expect(wrapper.contains('button')).toBe(true)
    })
    it("模拟按钮", async () => {
        expect(wrapper.vm.count).toBe(0)
        const button = wrapper.find('button')
        await button.trigger('click')
        expect(wrapper.text()).toContain('1')
    })
})

describe('ParentComponent父子组件', () => {
    it("displays 'Emitted!' when custom event is emitted", async () => {
        const wrapper = mount(ParentComponent)
        wrapper.find(ChildComponent).vm.$emit('custom')
        // wrapper.setData({ emitted: true })
        //  expect(wrapper.vm.emitted).toBe(true)
        await wrapper.vm.$nextTick()
        expect(wrapper.html()).toContain('Emitted!')
    })
})

const transitionStub = () => ({
    render: function (h) {
        return this.$options._renderChildren
    }
})


test('should render Foo, then hide it', async () => {
    const wrapper = mount(Foo, {
        stubs: {
            transition: transitionStub()
        }
    })
    expect(wrapper.text()).toMatch(/Foo/)

    wrapper.setData({
        show: false
    })
    await wrapper.vm.$nextTick()

    expect(wrapper.text()).not.toMatch(/Foo/)
})
test('should render Foo', async () => {
    const wrapper = mount(Foo, {
        data() {
            return {
                show: true
            }
        }
    })
    expect(wrapper.text()).toMatch(/Foo/)
})

test('should not render Foo', async () => {
    const wrapper = mount(Foo, {
        data() {
            return {
                show: false
            }
        }
    })
    expect(wrapper.text()).not.toMatch(/Foo/)
})

const $route = {
    path: '/',
    hash: '',
    params: { id: '123' },
    query: { q: 'hello' }
}
mount(Foo, {
    mocks: {
        $route
    }
})
*/
describe('Click event', () => {
    it('Click on yes button calls our method with argument "yes"', () => {
        const spy = sinon.spy()
        const wrapper = mount(testButton, {
            propsData: {
                callMe: spy
            }
        })
        wrapper.find('button.yes').trigger('click')


        expect(spy.calledWith('yes')).toBe(true)
    })
})

describe('Key event tests', () => {
    it('Quantity is zero by default', () => {
        const wrapper = mount(keyDown)
        expect(wrapper.vm.quantity).toBe(0)
    })

    it('Up arrow key increments quantity by 1', () => {
        const wrapper = mount(keyDown)
        wrapper.trigger('keydown.up')
        expect(wrapper.vm.quantity).toBe(1)
    })
    it('Down arrow key decrements quantity by 1', () => {
        const wrapper = mount(keyDown)
        wrapper.vm.quantity = 5
        wrapper.trigger('keydown.down')
        expect(wrapper.vm.quantity).toBe(4)
    })

    it('Escape sets quantity to 0', () => {
        const wrapper = mount(keyDown)
        wrapper.vm.quantity = 5
        wrapper.trigger('keydown.esc')
        expect(wrapper.vm.quantity).toBe(0)
    })

    it('Magic character "a" sets quantity to 13', () => {
        const wrapper = mount(keyDown)
        wrapper.trigger('keydown', {
            key: 'a'
        })
        expect(wrapper.vm.quantity).toBe(13)
    })
})
jest.mock('axios', () => ({
    get: Promise.resolve('value')
}))

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

shallowMount(mocks, {
    localVue,
    router
})
shallowMount(mocks, {
    stubs: ['router-link', 'touter-view']
})

const $route = {
    path: '/some/path'
}
const wrapper = shallowMount(mocks, {
    mocks: {
        $route
    }
})
wrapper.vm.$route.path