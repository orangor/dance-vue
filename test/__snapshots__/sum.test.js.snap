// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  
    0
    
  <button>
    自增
  </button>
</div>
`;
